lib/statsd/instrument/datagram_builder.rb

```diff
@@ -88,27 +126,7 @@ def generate_generic_datagram(name, value, type, sample_rate, tags)
       end
 
       def compile_tags(tags, buffer = "".b)
-        if tags.is_a?(Hash)
-          first = true
-          tags.each do |key, value|
-            if first
-              first = false
-            else
-              buffer << ","
-            end
-            key = key.to_s
-            key = key.tr("|,", "") if /[|,]/.match?(key)
-            value = value.to_s
-            value = value.tr("|,", "") if /[|,]/.match?(value)
-            buffer << key << ":" << value
-          end
-        else
-          if tags.any? { |tag| /[|,]/.match?(tag) }
-            tags = tags.map { |tag| tag.tr("|,", "") }
-          end
-          buffer << tags.join(",")
-        end
-        buffer
+        DatagramBuilder.normalize_tags(tags, buffer)
```

Comment:
If the method is just a delegator, might as well get rid of it and call the original one directly.

Reply:
I'm afraid I don't understand the question.

Reply:
The DatagramBuilder had a `protected` method `compile_tags` which comes from the "abstract" class StatsDDatagramBuilder. I wanted a `public` method that does exactly what `compile_tags` does. 

Should I just make the method public in `StatsDDatagramBuilder`?

Reply:
I am going with the solution of creating a new alias for now.

Reply:
I don't get why you need a public one but ok.

Reply:
I am using this from the Aggregator to create the key, so I need to share it. We can iterate later. Thx for all the help and feedback